packages/wasm-utils/src/note/note.rs

```diff
@@ -204,44 +172,132 @@ pub struct NoteBuilder {
 	pub group_id: u32,
 }
 
-impl NoteBuilder {
-	/*	fn get_net_generator(&self) -> Result<dyn Hasher + NoteGenerator, ()> {
-		match (self.backend, self.curve, self.hash_function) {
-			(Backend::Bulletproofs, ..) => {
-				let opts = PoseidonHasherOptions::default();
+struct NoteManager;
+
+impl NoteManager {
+	fn generate(note_builder: &NoteBuilder) -> Result<Note, ()> {
+		match (note_builder.backend, note_builder.curve) {
+			(Backend::Bulletproofs, Curve::Curve25519) => {
+				/*				let opts = PoseidonHasherOptions::default();
 				let pc_gens = PedersenGens::default();
 				let bp_gens = opts.bp_gens.clone().unwrap_or_else(|| BulletproofGens::new(16_400, 1));
 
 				let poseidon_hasher = PoseidonBuilder::new(opts.width)
-					.sbox(PoseidonSbox::Exponentiation3)
-					.bulletproof_gens(bp_gens)
-					.pedersen_gens(pc_gens)
-					.build();
+				  .sbox(PoseidonSbox::Exponentiation3)
+				  .bulletproof_gens(bp_gens)
+				  .pedersen_gens(pc_gens)
+				  .build();
 
-				let poseidon_note_generator = PoseidonNoteGeneratorCurve25519 {
-					hasher: poseidon_hasher,
+				let note_generator = PoseidonNoteGeneratorCurve25519 {
+				  hasher: poseidon_hasher,
 				};
-				Ok(poseidon_note_generator)
+				Ok(note_generator.generate(&self, &mut note_generator.get_rng()).unwrap())*/
+				unimplemented!()
 			}
 			(Backend::Circom, ..) => {
 				unimplemented!();
 			}
 			(Backend::Arkworks, Curve::Bn254) => {
-				unimplemented!();
+				let note_generator = match note_builder.hash_function {
+					HashFunction::Poseidon3 => ArkworksPoseidonBn254NoteGenerator::new(3, 3),
+					HashFunction::Poseidon5 => ArkworksPoseidonBn254NoteGenerator::new(5, 3),
+					HashFunction::Poseidon17 => ArkworksPoseidonBn254NoteGenerator::new(17, 3),
+					HashFunction::MiMCTornado => {
+						unreachable!()
+					}
+				};
+				Ok(note_generator
+					.generate(&note_builder, &mut note_generator.get_rng())
+					.unwrap())
 			}
 			(Backend::Arkworks, Curve::Bls381) => {
+				let note_generator = match note_builder.hash_function {
+					HashFunction::Poseidon3 => ArkworksPoseidonBls12_381NoteGenerator::new(3, 3),
+					HashFunction::Poseidon5 => ArkworksPoseidonBls12_381NoteGenerator::new(5, 3),
+					HashFunction::Poseidon17 => ArkworksPoseidonBls12_381NoteGenerator::new(17, 3),
+					HashFunction::MiMCTornado => {
+						unreachable!()
+					}
+				};
+				Ok(note_generator
+					.generate(&note_builder, &mut note_generator.get_rng())
+					.unwrap())
+			}
+			_ => {
+				dbg!(note_builder);
+				unimplemented!()
+			}
+		}
+	}
+
+	fn get_leaf_commitment(
+		backend: Backend,
+		curve: Curve,
+		hash_function: HashFunction,
+		secrets: &[u8],
+	) -> Result<Vec<u8>, ()> {
+		match (backend, curve) {
+			(Backend::Bulletproofs, Curve::Curve25519) => {
+				/*				let opts = PoseidonHasherOptions::default();
```

Comment:
I think if we choose to comment this out, it's best to remove bulletproofs entirely from the system. Otherwise, we should uncomment and ensure that this flow works still so that code doesn't go stale.